core/src/main/java/com/yahoo/imapnio/async/internal/ImapAsyncSessionImpl.java

```diff
@@ -188,21 +188,31 @@ public void setDebugMode(@Nonnull final DebugMode newOption) {
             throw new ImapAsyncClientException(FailureType.COMMAND_NOT_ALLOWED);
         }
 
-        final ImapFuture<ImapAsyncResponse> cmdFuture = new ImapFuture<ImapAsyncResponse>();
-        requestsQueue.add(new ImapCommandEntry(command, cmdFuture));
-
-        final ByteBuf buf = Unpooled.buffer();
-        final String tag = getNextTag();
-        buf.writeBytes(tag.getBytes(StandardCharsets.US_ASCII));
-        buf.writeByte(SPACE);
-        buf.writeBytes(command.getCommandLineBytes());
-
-        if (isDebugEnabled() && command.isCommandLineDataSensitive()) { // if we cannot log data sent over wire, ask command to provide log info
-            logger.debug(CLIENT_LOG_REC, sessionId, command.getDebugData());
+        try {
+            final ImapFuture<ImapAsyncResponse> cmdFuture = new ImapFuture<ImapAsyncResponse>();
+            requestsQueue.add(new ImapCommandEntry(command, cmdFuture));
+
+            final ByteBuf buf = Unpooled.buffer();
+            final String tag = getNextTag();
+            buf.writeBytes(tag.getBytes(StandardCharsets.US_ASCII));
+            buf.writeByte(SPACE);
+            buf.writeBytes(command.getCommandLineBytes());
+
+            if (isDebugEnabled() && command.isCommandLineDataSensitive()) { // if we cannot log data sent over wire, ask command to provide log info
+                logger.debug(CLIENT_LOG_REC, sessionId, command.getDebugData());
+            }
+            sendRequest(buf, command.isCommandLineDataSensitive());
+
+            return cmdFuture;
+        } catch (final RuntimeException e) {
```

Comment:
@jui8wang , should we catch all RuntimeException? Why is channel throwing NPE?

Reply:
I do catch RuntimeException, do u mean Exception?
Channel does not throw NPE, it is the getCommandLineBytes() in command that client can implement and write anything they like, which throws NPE.

Reply:
According to this: https://github.com/yahoo/imapnio/blob/master/core/src/main/java/com/yahoo/imapnio/async/request/ImapRequest.java, shouldn't getCommandLineInBytes be not null.